### PR TITLE
Make CMake find_package work for preinstalled libprotobuf on Raspbian.

### PR DESCRIPTION
**Pull request by kskalski**

### CODE REVIEW

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by jtattermusch** [comment]
test this please

**Comment by jtattermusch** [comment]
test this please

File: templates/CMakeLists.txt.template [hunk 1]
```diff
@@ -97,6 +97,9 @@
   set(gRPC_PROTOBUF_PROVIDER "module" CACHE STRING "Provider of protobuf library")
   set_property(CACHE gRPC_PROTOBUF_PROVIDER PROPERTY STRINGS "module" "package")
 
+  set(gRPC_PROTOBUF_PACKAGE_TYPE "" CACHE STRING "Algorithm for searching protobuf package")
```
Comment (jtattermusch) [inline templates/CMakeLists.txt.template]: Do we actually need this? It looks like that currently, we first try find_package(protobuf CONFIG) and then fall back to find_package(Protobuf MODULE) if not found, so I am not sure how helpful gRPC_PROTOBUF_PACKAGE_TYPE is.

Comment (kskalski) [inline templates/CMakeLists.txt.template]: Yes, this was my thinking too, but @podsvirov insisted on handling it manually. At the moment I have no idea when and on which platforms the CONFIG variant is useful, so I rely on @podsvirov 's insight.

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/11676
https://github.com/grpc/grpc/issues/11116
https://github.com/grpc/grpc/issues/11677

**Pull request merged**